File: solutions/0021-merge-two-sorted-lists/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2)
    {
        List<Integer>ls=new ArrayList<>();
        while(list1!=null)
        {
            ls.add(list1.val);
            list1=list1.next;
        }
        while(list2!=null)
        {
            ls.add(list2.val);
            list2=list2.next;
        }
        Collections.sort(ls);

        ListNode dummy = new ListNode(0);
        ListNode current = dummy;

        // Convert sorted ArrayList back to a linked list
        for (int value : ls) {
            current.next = new ListNode(value);
            current = current.next;
        }
        return dummy.next;    
    }
}

File: solutions/0021-merge-two-sorted-lists/solution.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* mergeTwoLists(struct ListNode* list1, struct ListNode* list2) {
    struct ListNode dummy;
    dummy.next = NULL;
    struct ListNode *res = &dummy;

    while(list1 && list2)
    {
        if(list1->val>list2->val)
        {
            res->next=list2;
            res=res->next;
            list2=list2->next;
        }
        else
        {
            res->next=list1;
            res=res->next;
            list1=list1->next;
        }
    }
    if(list1)
        res->next=list1;
    else
        res->next=list2;
    return dummy.next;
}
